BooksTest/app/src/main/java/com/rshtukaraxondevgroup/bookstest/database/AppDatabase.java

```diff
@@ -0,0 +1,31 @@
+package com.rshtukaraxondevgroup.bookstest.database;
+
+import android.arch.persistence.room.Database;
+import android.arch.persistence.room.Room;
+import android.arch.persistence.room.RoomDatabase;
+import android.content.Context;
+
+import com.rshtukaraxondevgroup.bookstest.model.BookModel;
+
+@Database(entities = {BookModel.class}, version = 1)
+public abstract class AppDatabase extends RoomDatabase {
+    public abstract BookDao bookDao();
+
+    private static String DATABASE = "database";
+
+    private static AppDatabase INSTANCE;
+
+    public static AppDatabase getDatabase(final Context context) {
+        if (INSTANCE == null) {
+            synchronized (AppDatabase.class) {
+                if (INSTANCE == null) {
```

Comment:
Do we really need to check instance twice?

Reply:
Yes, we do. You could find solution here -> https://medium.com/@oznusem/the-right-way-to-write-a-singleton-when-developing-for-android-or-any-multithreaded-environment-79782dd48f95

Reply:
If you use this approach, you'd better add 'volatile' keyword to instance.